Makie/src/stats/violin.jl

```diff
@@ -51,16 +51,14 @@ function getuniquevalue(v::AbstractVector, idxs)
 end
 
 function plot!(plot::Violin)
-    x, y = plot[1], plot[2]
-    args = @extract plot (
-        width, side, scale, color, show_median, npoints, boundary, bandwidth, weights,
-        datalimits, max_density, dodge, n_dodge, gap, dodge_gap, orientation,
-    )
-    signals = lift(
-        plot, x, y,
-        args...
-    ) do x, y, width, vside, scale_type, color, show_median, n, bound, bw, w, limits, max_density,
-            dodge, n_dodge, gap, dodge_gap, orientation
+    map!(
+        plot, [
+            :x, :y, :width, :side, :scale, :color, :show_median, :npoints, :boundary, :bandwidth,
+            :weights, :datalimits, :max_density, :dodge, :n_dodge, :gap, :dodge_gap, :orientation,
+        ],
+        [:vertices, :lines, :colors]
+    ) do x, y, width, vside, scale_type, color, show_median, n, bound, bw, w, limits,
+            max_density, dodge, n_dodge, gap, dodge_gap, orientation
```

Comment:
This also seems like a function that should be split up. I'd imagine `compute_x_and_width()` and `finduniquesorted` + `KernelDensity.kde` are probably relatively expensive and should get triggered unnecessarily, i.e. be in two isolated steps. The color stuff can probably be separated again.